.github/CODEOWNERS

```diff
@@ -30,15 +30,9 @@
 # /cpp/
 /cpp/src/arrow/acero @westonpace
 /cpp/src/arrow/adapters/orc @wgtmac
-/cpp/src/arrow/dataset @westonpace
 /cpp/src/arrow/engine @westonpace
 /cpp/src/arrow/flight/ @lidavidm
-/cpp/src/arrow/util/async* @westonpace
-/cpp/src/arrow/util/future* @westonpace
-/cpp/src/arrow/util/thread* @westonpace
 /cpp/src/parquet @wgtmac
-/cpp/src/skyhook @westonpace
-/csharp/ @westonpace
```

Comment:
Feel free to add me under /csharp so there's not nobody there.

Reply:
Done.  If you come across any C# PR that you would like a second opinion on then feel free to ping me / manually add me to review.